Automattic/jetpack
# Decide CSS files to proxy on the client-side

During Critical CSS generation, decide which CSS files need to be proxied on the client-side, to catch URL changes in WAFs for CDNs, etc.

Fixes Critical CSS Generation issues on Pressable sites.

Note: this PR refers to a branch in the Critical CSS Generation branch, to include this necessary PR: https://github.com/Automattic/jetpack-boost-critical-css-gen/pull/23 . Ensure that branch gets tagged as a release, and the package.json gets updated to the appropriate release tag before merging.

#### Changes proposed in this Pull Request:
* Updates Critical CSS Gen to a version which supports fetch overriding
* Overrides `fetch` when downloading CSS files to proxy CSS files from different origins.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Generate Critical CSS on a site with external CSS files
* Ensure local CSS files are not proxied, remote CSS files are proxied.
* Ensure the Proxy AJAX endpoint can't be abused.
